configs/components/openssl-1.1.1-fips.rb

```diff
@@ -27,26 +27,36 @@
   # proper fix would be extension in vanagon for source rpm handling
   pkg.add_source 'file://resources/patches/openssl/openssl-1.1.1-fips-patch-openssl-cnf.patch'
   pkg.add_source 'file://resources/patches/openssl/openssl-1.1.1-fips-force-fips-mode.patch'
-  pkg.add_source 'file://resources/patches/openssl/openssl-1.1.1-fips-post-rand.patch'
   pkg.add_source 'file://resources/patches/openssl/openssl-1.1.1-fips-spec-file.patch'
   pkg.add_source 'file://resources/patches/openssl/openssl-1.1.1-fips-remove-env-check.patch'
-  pkg.add_source 'file://resources/patches/openssl/openssl-1.1.1-fips-edk2-build.patch'
   pkg.add_source 'file://resources/patches/openssl/openssl-1.1.1l-ec-group-new-from-ecparameters.patch'
   pkg.add_source 'file://resources/patches/openssl/openssl-1.1.1l-sm2-plaintext.patch'
 
+  if platform.name =~ /-7-/
+    pkg.add_source 'file://resources/patches/openssl/openssl-1.1.1-fips-post-rand.patch'
```

Comment:
This patch removes the `random.h` include which seems to fail on redhat8fips, so avoid patching it there. The patch below just accounts for the changes in this patch so it's not needed anymore either.

```
crypto/rand/rand_unix.c: In function 'syscall_random':
crypto/rand/rand_unix.c:402:57: error: 'GRND_NONBLOCK' undeclared (first use in this function); did you mean 'O_NONBLOCK'?
     return syscall(SYS_getrandom, buf, buflen, nonblock?GRND_NONBLOCK:0);
                                                         ^~~~~~~~~~~~~
                                                         O_NONBLOCK
```